devices/third_reality.js

```diff
@@ -136,4 +136,17 @@ module.exports = [
             device.save();
         },
     },
+    {
+        zigbeeModel: ['3RTHS24BZ'],
+        model: '3RTHS24BZ',
+        vendor: 'Third Reality',
+        description: 'Temperature and humidity sensor',
+        fromZigbee: [fz.battery, fz.temperature, fz.humidity],
+        toZigbee: [],
+        exposes: [e.battery(), e.temperature(), e.humidity(), e.battery_voltage()],
+        configure: async (device, coordinatorEndpoint, logger) => {
+            device.powerSource = 'Battery';
```

Comment:
Could you post up the database.db entry for this device? I'd be very surprised if it doesn't report it's power source on interview.. If it does, you wont need this entry

Reply:
It will report the power.

Reply:
he is talking about the `powerSource`. When you force remove the device, restart z2m and pair the device; what power source shows up in the z2m frontend?

Reply:
Can you still check this @weihuan1111 ?

Reply:
![image](https://user-images.githubusercontent.com/99949392/194699396-eca27cb8-af79-4e0b-8aeb-307654b38199.png)

Reply:
![4293c425ca1f7229fe0723fe982e658](https://user-images.githubusercontent.com/99949392/194699419-c456fd1a-f515-4053-85b1-0595686ec034.jpg)

Reply:
![dcfdc5968f1cd6f815e3e5366cb7b3e](https://user-images.githubusercontent.com/99949392/194699433-e5523688-2bf5-4e5e-85c1-f7282875ed5c.jpg)

Reply:
@Koenkk You can see the above picture. I have tested it and it will display information such as power

Reply:
Could you post up your database.db entry as requested?

Reply:
database.db? How to post up database.db?

Reply:
You'll find it in the config folder where you installed Z2M

Reply:
![b63385ef7b53941731f0d75e43cea5a](https://user-images.githubusercontent.com/99949392/194733446-44e7ea01-ad1b-4737-bfde-5b9107348a1e.jpg)

Reply:
{"id":4,"type":"EndDevice","ieeeAddr":"0xa4c13847b1bede6c","nwkAddr":12222,"manufId":4659,"manufName":"Third Reality, Inc","powerSource":"Battery","modelId":"3RTHS24BZ","epList":[1],"endpoints":{"1":{"profId":260,"epId":1,"devId":770,"inClusterList":[0,1,1026,1029],"outClusterList":[25],"clusters":{"msTemperatureMeasurement":{"attributes":{"measuredValue":2800}},"msRelativeHumidity":{"attributes":{"measuredValue":6500}},"genBasic":{"attributes":{"manufacturerName":"Third Reality, Inc","powerSource":1,"zclVersion":3,"swBuildId":"1.00.14","appVersion":14,"stackVersion":2,"hwVersion":0}},"genPowerCfg":{"attributes":{"batteryVoltage":30,"batteryPercentageRemaining":200}}},"binds":[],"configuredReportings":[],"meta":{}}},"appVersion":14,"stackVersion":2,"hwVersion":0,"swBuildId":"1.00.14","zclVersion":3,"interviewCompleted":true,"meta":{"configured":1955752412},"lastSeen":1664521072539,"defaultSendRequestWhen":"immediate"}

Reply:
`"powerSource":"Battery"` looks good so it is not needed. I will remove it.

Reply:
What is the function of the paragraph you deleted？

Reply:
It sets the correct power source of the device in Z2M if it is not reported by the device itself during interview